src/Atmos/Model/AtmosModel.jl

```diff
@@ -1141,4 +1149,243 @@ function numerical_flux_first_order!(
+    balance_law.moisture isa EquilMoist ||
+        error("Must use a EquilMoist model for RoeNumericalFluxMoist")
+    numerical_flux_first_order!(
+        CentralNumericalFluxFirstOrder(),
+        balance_law,
+        fluxᵀn,
+        normal_vector,
+        state_prognostic⁻,
+        state_auxiliary⁻,
+        state_prognostic⁺,
+        state_auxiliary⁺,
+        t,
+        direction,
+    )
+
+    FT = eltype(fluxᵀn)
+    param_set = balance_law.param_set
+    _cv_d::FT = cv_d(param_set)
+    _T_0::FT = T_0(param_set)
+    γ::FT = cp_d(param_set) / cv_d(param_set)
+    _e_int_v0::FT = e_int_v0(param_set)
+    Φ = gravitational_potential(balance_law, state_auxiliary⁻)
+
+    ρ⁻ = state_prognostic⁻.ρ
+    ρu⁻ = state_prognostic⁻.ρu
+    ρe⁻ = state_prognostic⁻.ρe
+    ρq_tot⁻ = state_prognostic⁻.moisture.ρq_tot
+
+    u⁻ = ρu⁻ / ρ⁻
+    e⁻ = ρe⁻ / ρ⁻
+    ts⁻ = recover_thermo_state(balance_law, state_prognostic⁻, state_auxiliary⁻)
+    h⁻ = total_specific_enthalpy(ts⁻, e⁻)
+    qt⁻ = ρq_tot⁻ / ρ⁻
+    c⁻ = soundspeed_air(ts⁻)
+
+    ρ⁺ = state_prognostic⁺.ρ
+    ρu⁺ = state_prognostic⁺.ρu
+    ρe⁺ = state_prognostic⁺.ρe
+    ρq_tot⁺ = state_prognostic⁺.moisture.ρq_tot
+
+    u⁺ = ρu⁺ / ρ⁺
+    e⁺ = ρe⁺ / ρ⁺
+    ts⁺ = recover_thermo_state(balance_law, state_prognostic⁺, state_auxiliary⁺)
+    h⁺ = total_specific_enthalpy(ts⁺, e⁺)
+    qt⁺ = ρq_tot⁺ / ρ⁺
+    c⁺ = soundspeed_air(ts⁺)
+    ũ = roe_average(ρ⁻, ρ⁺, u⁻, u⁺)
+    e_tot = roe_average(ρ⁻, ρ⁺, e⁻, e⁺)
+    h̃ = roe_average(ρ⁻, ρ⁺, h⁻, h⁺)
+    qt = roe_average(ρ⁻, ρ⁺, qt⁻, qt⁺)
+    ρ = sqrt(ρ⁻ * ρ⁺)
+    e_int⁻ = internal_energy(ts⁻)
+    e_int⁺ = internal_energy(ts⁺)
+    e_int = roe_average(ρ⁻, ρ⁺, e_int⁻, e_int⁺)
+    ts = PhaseEquil(
+        param_set,
+        e_int,
+        ρ,
+        qt,
+        balance_law.moisture.maxiter,
+        balance_law.moisture.tolerance,
+    )
+    c̃ = sqrt((γ - 1) * (h̃ - (ũ[1]^2 + ũ[2]^2 + ũ[3]^2) / 2))
+    (R_m, _cp_m, _cv_m, gamma) = gas_constants(ts)
+    # chosen by fair dice roll
+    # guaranteed to be random
+    ω = FT(π) / 3
+    δ = FT(π) / 5
+    random_unit_vector = SVector(sin(ω) * cos(δ), cos(ω) * cos(δ), sin(δ))
+    # tangent space basis
+    τ1 = random_unit_vector × normal_vector
+    τ2 = τ1 × normal_vector
+    ũᵀn⁻ = u⁻' * normal_vector
+    ũᵀn⁺ = u⁺' * normal_vector
+    ũᵀn = ũ' * normal_vector
+    ũc̃⁻ = ũ + c̃ * normal_vector
+    ũc̃⁺ = ũ - c̃ * normal_vector
+    e_kin_pot = h̃ - _e_int_v0 * qt - _cp_m * c̃^2 / R_m
+    if (numerical_flux.LM == true)
+        Mach⁺ = sqrt(u⁺' * u⁺) / c⁺
+        Mach⁻ = sqrt(u⁻' * u⁻) / c⁻
+        Mach = (Mach⁺ + Mach⁻) / 2
+        c̃_LM = c̃ * min(Mach * sqrt(4 + (1 - Mach^2)^2) / (1 + Mach^2), 1)
+    else
+        c̃_LM = c̃
+    end
+    #Standard Roe
+    Λ = SDiagonal(
+        abs(ũᵀn - c̃_LM),
+        abs(ũᵀn),
+        abs(ũᵀn),
+        abs(ũᵀn),
+        abs(ũᵀn + c̃_LM),
+        abs(ũᵀn),
+    )
+    #Harten Hyman
+    if (numerical_flux.HH == true)
```

Comment:
@slifer50 just to note, it might be an improvement to use dispatch for these, rather than `Bool`s. But the method interface might not be the same and some parts are needed for multiple numerical fluxes, so I couldn't see a clear win at first glance.

Reply:
Yeah, I'm not a big fan of the current way this is being handled, but I don't really know a simple way to do this without building these as entirely separate fluxes.

Reply:
We can try to do that, if it's worthwhile, in a follow-up PR

Reply:
I agree with this.